### PR TITLE
Fixes being unable to ignite bonfire in lavaland

### LINKED ISSUE #26642: Cannot Light Bonfire on Lavaland
### Exploit Reports

- [X] I confirm this issue is not an exploit. (Required)

### BYOND Version

515.1634.0

### Issue Description

Lighting the bonfire at the Ash Walker camp on Lavaland is not possible.

### What did you expect to happen?

To be able to light the bonfire at the Ash Walker camp on Lavaland.

### What happened instead?

The bonfire did not light.

### Why is this bad/What are the consequences?

Being able to light a fire on Lavaland as an Ash Walker is important as a source of light in the Ash Walker spawn area.

### Steps to reproduce the issue.

Spawn in as an Ash Walker.
Use the provided tools (welder) to try to light a fire.
Watch as nothing happens.

### When did the problem start happening?

Unknown. Possibly when MILLA was implemented. 

### Extra information

_No response_

### Relevant log output/runtime error

_No response_

### PR DESCRIPTION
**Pull request by kano-dot**
## What Does This PR Do
Changes required minimum oxygen mole value to ignite a bonfire from `14` to `8` (each lavaland tile has 8 moles of oxygen) and adds a notification message when there's not enough oxygen. Fixes #26642.

## Why It's Good For The Game

This is likely an oversight. Players should be able to ignite bonfires in lavaland again.

## Testing

Tried to ignite it in both station and lavaland wastes, it burned. Tried again in space vacuum, it didn't burn.

<hr>

### Declaration

- [x] I confirm that I either do not require [pre-approval](https://github.com/ParadiseSS13/Paradise/blob/master/CODE_OF_CONDUCT.md#types-of-changes-that-need-approval) for this PR, or I have obtained such approval and have included a screenshot to demonstrate this below.
  
  
  <hr>

## Changelog

:cl:
fix: Fixed being unable to ignite bonfires in lavaland.
/:cl:

### CODE REVIEW

File: code/modules/hydroponics/grown/towercap.dm [hunk 1]
```diff
@@ -181,17 +181,19 @@
...
-/obj/structure/bonfire/proc/StartBurning()
+/obj/structure/bonfire/proc/StartBurning(mob/user)
 	if(!burning && CheckOxygen())
 		icon_state = "bonfire_on_fire"
 		burning = 1
 		set_light(6, l_color = "#ED9200")
 		Burn()
 		START_PROCESSING(SSobj, src)
+	if(!CheckOxygen())
+		to_chat(user, "<span class='warning'>You can't seem to ignite [src] in this environment!")
```
Comment (CinnamonSnowball) [inline code/modules/hydroponics/grown/towercap.dm:196]:
```suggestion
	if(burning)
		return
	if(!CheckOxygen())
		to_chat(user, "<span class='warning'>You can't seem to ignite [src] in this environment!")
		return

	icon_state = "bonfire_on_fire"
	burning = TRUE
	set_light(6, l_color = "#ED9200")
	Burn()
	START_PROCESSING(SSobj, src)
```

**Comment by Scribble-Sheep** [comment]
Wait, so *that's* what the issue was?
Lmao this was so embedded that even admin-spawned prelit ones were broken.

File: code/modules/hydroponics/grown/towercap.dm [hunk 2]
```diff
@@ -181,17 +181,22 @@
...
-		icon_state = "bonfire_on_fire"
-		burning = 1
-		set_light(6, l_color = "#ED9200")
-		Burn()
-		START_PROCESSING(SSobj, src)
+/obj/structure/bonfire/proc/StartBurning(mob/user)
+	if(burning)
+		return
+	if(!CheckOxygen())
+		to_chat(user, "<span class='warning'>You can't seem to ignite [src] in this environment!")
```
Comment (BiancaWilkson) [inline code/modules/hydroponics/grown/towercap.dm:192]: Make sure to close this span.

File: code/modules/hydroponics/grown/towercap.dm [hunk 3]
```diff
@@ -181,17 +181,22 @@
 /obj/structure/bonfire/proc/CheckOxygen()
 	var/turf/T = get_turf(src)
 	var/datum/gas_mixture/G = T.get_readonly_air()
-	if(G.oxygen() > 13)
+	if(G.oxygen() > 7)
```
Comment (lewcc) [inline code/modules/hydroponics/grown/towercap.dm:184]: Is there a define for this value anywhere?

Comment (kano-dot) [reply code/modules/hydroponics/grown/towercap.dm:184]: nothing that I could find. Should I make one in this file?

Comment (S34NW) [reply code/modules/hydroponics/grown/towercap.dm:184]: please do, we hate magic numbers

File: code/modules/hydroponics/grown/towercap.dm [hunk 4]
```diff
@@ -154,18 +157,18 @@
...
 		buckle_requires_restraints = TRUE
-		to_chat(user, "<span class='italics'>You add a rod to [src].")
+		to_chat(user, "<span class='italics'>You add a rod to [src].</span>")
 		var/image/U = image(icon='icons/obj/hydroponics/equipment.dmi',icon_state="bonfire_rod",pixel_y=16)
 		underlays += U
 	if(W.get_heat())
 		lighter = user.ckey
 		user.create_log(MISC_LOG, "lit a bonfire", src)
-		StartBurning()
+		StartBurning(user)
```
Comment (DGamerL) [inline code/modules/hydroponics/grown/towercap.dm:167]:
```suggestion
```

**Review by DGamerL — APPROVED** [review]

**Review by Burzah — APPROVED** [review]

**Pull request merged**